packages/project/src/utils/v1-utils.ts

```diff
@@ -153,13 +157,80 @@ const checkEventDates = (
   return filteredCrop;
 };
 
+// eslint-disable-next-line jsdoc/require-example
+/**
+ * The very first crop year on a field is subject to a problem where any fertilizer or tillage events
+ * that occur in the calendar year PRIOR to the crop year (like fall fertilizer) will cause the import
+ * code to throw because there doesn't exist a crop in the corresponding year to assign the event to.
+ * We therefore notify the user of the problematic event and remove it from the import.
+ *
+ * @param crop The crop for which we are looking for edge-case event date problems.
+ * @param fieldName The name of the field currently being checked.
+ * @param earliestCropYear The year of the earliest crop year that appears in this field.
+ * @param errorCollector The error collector aggregating all errors for this import file.
+ *
+ * @returns The original crop but with offending events removed.
+ */
+const checkFertilizerTillageDateEdgeCase = (
+  crop: V1Crop,
+  fieldName: string,
+  earliestCropYear: number,
+  errorCollector: ErrorCollector
+): V1Crop => {
+  const filteredCrop = { ...crop };
+  const filteredFertilizerEvents: V1FertilizerEvent[] = [];
+  const filteredTillageEvents: V1TillageEvent[] = [];
+  crop.fertilizerEvents.forEach((fertilizerEvent) => {
```

Comment:
This works fine as is.  My functional programming brain itches to see it written in terms of sets and/or a map+filter or reduce operation.  i.e.

setOfBadOnes = events.filter(findBadOnes);
setOfBadOnes.foreach(pushTheError);
return events - setOfBadOnes;

Extreme liberties taken in that pseudo-code ;)

Reply:
```ts
filteredCrop.fertilizerEvents.filter((event) => {
  const eventDateIsBeforeEarliestCropYear = moment(event.date).isBefore(moment(earliestCropYear));
  if (eventDateIsBeforeEarliestCropYear) {
    errorCollector.collectKeyedError(
      'projectDataError:priorYearEdgeCaseError', 
      { field: fieldName, event }
    );
  }
  return !eventDateIsBeforeEarliestCropYear
});
```

Reply:
nice, thank you for making me a better JS developer!! took the liberty of applying a similar cleanup to the other event filters from an earlier PR.  Still some ugly code duplication there (should have handled the error context reporting for each event type more generically... but at this point I don't think it's worth tweaking all the test expectations just to refactor this piece since hopefully this code will go away at some point soon)